app/scripts/components/dashboard/charts/boreholes-stats-chart.jsx

```diff
@@ -0,0 +1,86 @@
+import React, { PropTypes } from 'react';
+import {LineChart} from 'react-d3-components';
+import * as func from '../../../utils/functional';
+import TSetChildProps from '../../misc/t-set-child-props';
+import T from '../../misc/t';
+
+require('stylesheets/dashboard/charts/boreholes-stats-chart');
+
+const BoreholesStatsChart = React.createClass({
+  propTypes: {
+    boreholes: PropTypes.array.isRequired,
+  },
+
+  getInitialState() {
+    return {
+      DIAMETER: {enabled: true, color: '#FA8072' },
+      DEPTH_METER: {enabled: true, color: '#FF4500' },
+      STATIC_WATER_LEVEL: {enabled: true, color: '#87CEEB' },
+      DYNAMIC_WATER_LEVEL_METER: {enabled: true, color: '#000080' },
+      'DRAW _DOWN_METER': {enabled: true, color: '#DDA0DD' },
+      YIELD_METER_CUBED_PER_HOUR: {enabled: true, color: '#4B0082' },
+    };
+  },
+
+  getActiveMetrics() {
+    return Object.keys(this.state)
+      .filter(metric => this.state[metric].enabled)
+      .map(metric => metric);
+  },
+
+  parseData(data) {
+    return this.getActiveMetrics().map(key => {
+      return {
+        label: key,
+        values: Object.keys(data)
+                  .filter(year => data[year])
+                  .map(year => {
+                    return {
+                      x: year,
+                      y: data[year].reduce((res, metric) => {
+                        if (metric[key]) {
+                          res.value = metric[key];
+                        }
+                        return res;
+                      }, {value: 0}).value,
+                      y0: 0,
+                    };
+                  }),
+      };
+    });
+  },
+
+  toogleMetric(e, metric) {
+    this.state[metric].enabled = !this.state[metric].enabled;
```

Comment:
Don't mutate `this.state`. In React it can make Bad Things Happen (TM).

To get a non-mutated version with a nested change, this is one way:
```js
const newState = {
  ...this.state,
  [this.state[metric]]: {
    ...this.state[metric],
    enabled: !this.state[metric].enabled,
  },
};
```

...though it's pretty hairy. It might be worth pulling in immutable.js for this, which has nice functions for immutably updating nested structures. There is also [deep extend](https://www.npmjs.com/package/deep-extend) which might work.